docs/styleguide.md

```diff
@@ -0,0 +1,24 @@
+
+List of programming style rules
+===============================
+
+(will be re-structured as it grows)
+
+- qualified imports are given the name of the module (without path),
+  not an abbreviation (`import qualified Data.Map as Map`).
+
+- exceptions:
+
+    import Data.ByteString as SBS
+    import Data.ByteString.Lazy as LBS
+    import Data.Text as ST
+    import Data.Text.Lazy as LT
+
+- type synonyms SBS, LBS, ST, LST from string-conversions are used in
+  type signatures (even if no strings are converted).
+
```

Comment:
I don't love this rule. I don't think it improves anything, but it makes signatures less readable by replacing well-known types by obscure and unreadable synonyms.